bookwyrm/activitypub/note.py

```diff
@@ -38,6 +41,35 @@ class Note(ActivityObject):
     updated: str = None
     type: str = "Note"
 
+    # pylint: disable=too-many-arguments
+    def to_model(
+        self, model=None, instance=None, allow_create=True, save=True, overwrite=True
+    ):
+        instance = super().to_model(model, instance, allow_create, save, overwrite)
+
+        # Replace links to hashtags in content with local URLs
+        changed_content = False
+        for hashtag in instance.mention_hashtags.all():
+            updated_content = re.sub(
+                rf'(<a href=")[^"]*/hashtag/[^"]*(">{hashtag.name}</a>)',
```

Comment:
This regex is really tailored to BookWyrm-style hashtags.

The status content from BookWyrm with a status looks like: `<p>Hello World <a href="https: //dev-lit.secretbearlibrary.org/hashtag/1">#BookClub</a></p>`

An example from Mastodon looks like: `<p>Hi <span class="h-card"><a href="https://dev-sci.secretbearlibrary.org/user/sciadmin" class="u-url mention">@<span>sciadmin</span></a></span> <a href="https: //secretbearsociety.org/tags/BookWyrm" class="mention hashtag" rel="tag">#<span>BookWyrm</span></a> is rad</p>`

Makes me think that maybe we should add something like `class="mention hashtag"` and `rel="tag"` as well. Writing a regex that can parse this out without relying on the order seems like a massive pain, but maybe that's overkill.

Reply:
Opted to go with a `data-mention="hashtag"` attribute on the link. Didn't feel like it justified opening up the `class` attribute in our HTML sanitizer, as it could be used to apply CSS styles to user-generated content.